embeddings/evaluator/metrics_evaluator.py

```diff
@@ -1,30 +1,44 @@
 import abc
-from typing import Any, Dict, List, Sequence, Union
+from typing import Any, Dict, Generic, List, Type, TypeVar, Union
 
 import torch
 from numpy import typing as nptyping
 
+from embeddings.evaluator.evaluation_results import EvaluationResults, Predictions
 from embeddings.evaluator.evaluator import Evaluator
 from embeddings.metric.metric import Metric
 
+EvaluationResultsType = TypeVar("EvaluationResultsType", bound=EvaluationResults)
 
-class MetricsEvaluator(Evaluator[Dict[str, nptyping.NDArray[Any]], Dict[str, Any]]):
+
+class MetricsEvaluator(
+    Evaluator[Predictions, EvaluationResultsType],
+    Generic[EvaluationResultsType],
+):
     def __init__(self, return_input_data: bool = True):
         super().__init__()
```

Comment:
the parent class `Evaluator.init` method is empty so we don't need to call `super.init()`